src/util/mod.rs

```diff
@@ -49,6 +51,34 @@ pub mod worker;
 pub use self::rocksdb::properties;
 pub use self::rocksdb::stats as rocksdb_stats;
 
+static mut PANIC_MARK: bool = false;
```

Comment:
How about changing it to a AtomicBool to avoid `unsafe` and set a correct MemoryOrdering to ensure the reading is not affected by cache?